### ISSUE TITLE
Make discretization-implied difference scheme default in derivatives

### ISSUE THREAD

**Issue by kohr-h** [odlgroup/odl#268]
By default, the discretization should imply which difference scheme to use in the partial derivative operators. Other options can be selected explicitly. See #133

**Comment by adler-j** [comment]
I disagree with this. Forward differences is "the standard" for regularization, and using central differences causes all kinds of very subtle errors. Further, this would require us to also add per-axis difference schemes (not terribly hard, but still).

**Comment by kohr-h** [comment]
Per-axis should definitely be added, also since it's [supported natively in Numpy](https://github.com/numpy/numpy/blob/master/doc/release/1.13.0-notes.rst#npgradient-now-supports-unevenly-spaced-data) now. About the default, yeah, I guess "practicality beats purity" applies here. So let's go with whatever works best in the most common use cases.
